oak_io/Cargo.toml

```diff
@@ -9,3 +9,9 @@ license = "Apache-2.0"
 
 [dependencies]
 oak_abi = { path = "../oak_abi" }
+
+[patch.crates-io]
+prost = { path = "../third_party/prost" }
+prost-build = { path = "../third_party/prost/prost-build" }
+prost-derive = { path = "../third_party/prost/prost-derive" }
+prost-types = { path = "../third_party/prost/prost-types" }
```

Comment:
@daviddrysdale I have added these patches to various crates, which actually we should have done in the past already; if you look at the diff in the corresponding Cargo.lock, the previous version used to depend on prost from github, while now it depends on the local vendored copy (which is expected).

The problem is that when we run `cargo fmt --all`, it tries to format all those third party crates too for whatever reason. Have you come across this issue before? Do you have any idea how to prevent it?

The main thing I can think of is to reverse the logic in runner so that it only runs `cargo fmt` on **non-workspace** Cargo.toml files, i.e. the opposite of what it does now. I guess it may make that step more fine grain and therefore possibly a bit slower, but I don't think it matters that much.

WDYT?

Reply:
Would `cargo fmt --all --exclude prost --exclude prost-build --exclude prost-derive --exclude prost-types` work?

Reply:
My version of `cargo fmt` does not seem to have an `--exclude` flag, does yours? If so, what version is it?

Reply:
Ah no, sorry &ndash; I didn't try it, I just followed https://github.com/rust-lang/cargo/issues/2878 & https://github.com/rust-lang/cargo/pull/4031 and assumed `--exclude` applied everywhere.

Reply:
Maybe create a `third_party/prost/.rustfmt.toml` file with an `ignore` directive that matches `*.rs` ?

Reply:
Great idea! I actually created it under `third_party` directly, and ignored everything. Thanks for the suggestion!